File: Python/7py 2/palindrome.py
"""
|--------------------------------------------------------------------------
| Numerical Palindrome #1
|--------------------------------------------------------------------------
|
"""

def palindrome(num):
   return str(num) == str(num)[::-1] if type(num) == int and num >= 0 else "Not valid"


def palindrome(n):
   try:
      return 1. * n == int(str(n)[::-1])
   except:
      return "Not valid"


def palindrome(num):
   return "Not valid" if not isinstance(num, int) or num < 0 else str(num) == ''.join(reversed(str(num))) 


print(palindrome(1221), True)
print(palindrome(123322), False)
print(palindrome("ACCDDCCA"), "Not valid")
print(palindrome("1221"), "Not valid")
print(palindrome(-450), "Not valid")

File: JavaScript/7js 2/palindrome.js
/*
|--------------------------------------------------------------------------
| Numerical Palindrome #1
|--------------------------------------------------------------------------
|
*/

function palindrome(num) {
   return typeof (num) == 'number' && num >= 0 ? String(num) == String(num).split('').reverse().join('') : "Not valid"
}


function palindrome(num) {
   return Number.isInteger(num) && num >= 0 ? parseInt(`${num}`.split("").reverse().join("")) === num : "Not valid"
}


function palindrome(num) {
   return Number.isInteger(num) && num >= 0 ? [...`${num}`].reverse().join(``) == num : `Not valid`;
}


console.log(palindrome(1221), true);
console.log(palindrome(123322), false);
console.log(palindrome("ACCDDCCA"), "Not valid");
console.log(palindrome("1221"), "Not valid");
console.log(palindrome(-450), "Not valid");